rust/ql/lib/codeql/rust/elements/internal/AstNodeImpl.qll

```diff
@@ -26,21 +26,29 @@ module Impl {
     result = getImmediateParent(e)
   }
 
-  /** Gets the nearest enclosing parent of `ast` that is an `AstNode`. */
-  private AstNode getParentOfAst(AstNode ast) {
-    result = getParentOfAstStep*(getImmediateParent(ast))
-  }
-
   class AstNode extends Generated::AstNode {
+    /**
+     * Gets the nearest enclosing parent of this node, which is also an `AstNode`,
+     * if any.
+     */
+    AstNode getParentNode() { result = getParentOfAstStep*(getImmediateParent(this)) }
```

Comment:
I'd prefer `getParent` . I think all immediate parents are actually AstNodes, and if not, we should probably change the schema.

Reply:
I can't be named `getParent` because it will then conflict with the auto-generated `Comment.getParent` predicate :-(